lib/server.js

```diff
@@ -437,28 +437,41 @@ Server.prototype._createAccessory = function(accessoryInstance, displayName, acc
       if (service instanceof Service.AccessoryInformation) {
         var existingService = accessory.getService(Service.AccessoryInformation);
 
-        // pull out any values you may have defined
-        var manufacturer = service.getCharacteristic(Characteristic.Manufacturer).value;
-        var model = service.getCharacteristic(Characteristic.Model).value;
-        var serialNumber = service.getCharacteristic(Characteristic.SerialNumber).value;
-        var firmwareRevision = service.getCharacteristic(Characteristic.FirmwareRevision).value;
-        var hardwareRevision = service.getCharacteristic(Characteristic.HardwareRevision).value;
-
-        if (manufacturer) existingService.setCharacteristic(Characteristic.Manufacturer, manufacturer);
-        if (model) existingService.setCharacteristic(Characteristic.Model, model);
-        if (serialNumber) existingService.setCharacteristic(Characteristic.SerialNumber, serialNumber);
-        if (firmwareRevision) existingService.setCharacteristic(Characteristic.FirmwareRevision, firmwareRevision);
-        if (hardwareRevision) existingService.setCharacteristic(Characteristic.HardwareRevision, hardwareRevision);
+        this._augmentService(service, existingService, Characteristic.Name);
+        this._augmentService(service, existingService, Characteristic.Manufacturer);
+        this._augmentService(service, existingService, Characteristic.Model);
+        this._augmentService(service, existingService, Characteristic.SerialNumber);
+        this._augmentService(service, existingService, Characteristic.FirmwareRevision);
+        this._augmentService(service, existingService, Characteristic.HardwareRevision);
       }
       else {
         accessory.addService(service);
       }
-    });
+    }.bind(this));
 
     return accessory;
   }
 }
 
+Server.prototype._augmentService = function(srcService, dstService, characteristicUUID) {
+  var srcCharacteristic = srcService.getCharacteristic(characteristicUUID);
+  var dstCharacteristic = dstService.getCharacteristic(characteristicUUID);
+
+  if (!srcCharacteristic) return;
```

Comment:
This will always return false, as "getCharacteristic" will create the characteristic if it is not present.